### PR TITLE
Add filters to schedule_e endpoint.

### LINKED ISSUE #3276: Add filters for candidate state, district, and party affiliation to /schedules/schedule_e/
Add 
`candidate_office`
`candidate_state`
`candidate_district`
`candidate_party` to /schedules/schedule_e/

- [ ] Add the filter to `args.py`
- [ ] Modify the query/MV as needed
- [ ] Add indexes as needed

### PR DESCRIPTION
**Pull request by pkfec**
## Summary (required)

- Resolves #3276

_Include a summary of proposed changes._

- Add candidate office,  state, district and party  filters on schedule_e endpoint.
- Update existing  test class `test_filters_sched_e` and test the four new filters on schedule_e endpoint.

Updated below files:
- webservices/resources/sched_e.py
- webservices/common/models/itemized.py
- webservices/args.py
- tests/test_itemized.py
- data/migrations/V0104__update_ofec_sched_e_mv_add_columns.sql

## How to test the changes locally
In openfec terminal:
-  git checkout feature/add-schedule-e-filters
-  change the  **tablename to 'ofec_sched_e_mv_2'** in the model file here:
https://github.com/fecgov/openFEC/blob/develop/webservices/common/models/itemized.py#L524
-  export SQLA variable to point to DEV DB
- ./manage.py runserver

In browser:
- open the local swagger api :  http://127.0.0.1:5000/developers
- test the 4 new filters on **schedules/schedule_e/** endpoint under **/independent expenditures/**  section 

## Impacted areas of the application
List general components of the application that this PR will affect:

- Independent Expenditure page

### CODE REVIEW

**Comment by pkfec** [comment]
@fec-jli and @lbeaufort Please review when you get a chance.

**Comment by pkfec** [comment]
@fec-jli @lbeaufort :  Please review

**Comment by lbeaufort** [comment]
@pkfec sorry I missed your tests - need more coffee. Just the small comments from me :)

**Comment by pkfec** [comment]
@lbeaufort thank you for taking time to review this PR. 
I am updated the existing test and included the 3 new filters here. Please take a look
https://github.com/fecgov/openFEC/pull/3348/files#diff-63af08d3af013aa096e9ad50558a5b41L631

File: webservices/common/models/itemized.py [hunk 1]
```diff
@@ -521,7 +521,7 @@ def pdf_url(self):
 
 
 class ScheduleE(PdfMixin, BaseItemized):
-    __tablename__ = 'ofec_sched_e'
+    __tablename__ = 'ofec_sched_e_mv'
```
[the remark this answers is not in the document]
Comment (pkfec) [reply webservices/common/models/itemized.py]: Yes.@Jean is going to drop and its part of another issue.

File: webservices/common/views.py [hunk 2]
```diff
@@ -9,6 +9,8 @@
 from webservices.common import models
 from webservices.utils import use_kwargs
 
+from sqlalchemy.dialects import postgresql
```
[the remark this answers is not in the document]
Comment (pkfec) [reply webservices/common/views.py]: No. i will remove this.

File: webservices/common/models/itemized.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (fecjjeng) [reply webservices/common/models/itemized.py]: Drop the current ofec_sched_e and related objects and processes are part of issue #3100 (drop ofec_sched_e table and related process).  After completion of issue #3089 (Switching API to the fecp-driven schedule table (sched_e)).

File: data/migrations/V0104__update_ofec_sched_e_mv_add_columns.sql [hunk 3]
```diff
@@ -0,0 +1,472 @@
...
+se.orig_sub_id,
+se.sub_id,
+'F24' AS filing_form,
+f24.rpt_tp,
+f24.rpt_yr,
+f24.rpt_yr + mod(f24.rpt_yr, 2::numeric) AS election_cycle,
+image_pdf_url(se.image_num) as pdf_url,
+coalesce(f24.rpt_tp, '') in ('24', '48') as is_notice,
+to_tsvector(se.pye_nm) as payee_name_text
+FROM disclosure.nml_sched_e se,
```
[the remark this answers is not in the document]
Comment (lbeaufort) [reply data/migrations/V0104__update_ofec_sched_e_mv_add_columns.sql]: GitHub only lets me comment on one line, but if it's more helpful here are the lines: 
https://github.com/fecgov/openFEC/blob/f98b36c530f6c2669609217aca9f7f042ba325e4/data/migrations/V0104__update_ofec_sched_e_mv_add_columns.sql#L182-L184
https://github.com/fecgov/openFEC/blob/f98b36c530f6c2669609217aca9f7f042ba325e4/data/migrations/V0104__update_ofec_sched_e_mv_add_columns.sql#L358-L360

Also, while you're in there can you please new-line and indent all the `AND` clauses?

[the remark this answers is not in the document]
Comment (fecjjeng) [reply data/migrations/V0104__update_ofec_sched_e_mv_add_columns.sql]: The comma separated list is a more traditional way.  When there are more tables join together, the newer JOIN syntax does add more clarity.  However, for this two table join, it does not matter much.  Nonetheless, I had updated the file according to the request.

File: webservices/common/models/itemized.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jean) [reply webservices/common/models/itemized.py]: @pkfec that's a different `jean` .. :wink:

**Pull request merged**